CMakeLists.txt

```diff
@@ -57,15 +56,38 @@ if (${LLVM_PACKAGE_VERSION} VERSION_LESS "5.0")
   message(FATAL_ERROR "LLVM version must be >= 5.0")
 endif()
 
+if (NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+  if (${LLVM_PACKAGE_VERSION} VERSION_LESS "10.0")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+  endif()
+endif()
+
 # Add flags for LLVM
 add_definitions(${LLVM_DEFINITIONS})
 include_directories(${LLVM_INCLUDE_DIRS})
 link_directories(${LLVM_LIBRARY_DIRS})
 
 # Get LLVM libraries for linking
-llvm_map_components_to_libnames(LLVM_LIBS
-  bitreader bitwriter core coroutines coverage instrumentation ipo irreader
-  linker lto mcparser objcarcopts option target)
+list(FIND LLVM_AVAILABLE_LIBS LLVM _LLVM_SHARED_INDEX)
+if (${_LLVM_SHARED_INDEX} GREATER -1)
+  set(LLVM_LIBS LLVM)
+else()
+  llvm_map_components_to_libnames(LLVM_LIBS
+    bitreader bitwriter core coroutines coverage instrumentation ipo irreader
+    linker lto mcparser objcarcopts option target)
+  if (${LLVM_PACKAGE_VERSION} VERSION_GREATER "9.0")
```

Comment:
This didn't work for me on a system with `9.0.1`. Can you make it `VERSION_GREATER_EQUAL "10.0"` please?

Reply:
`VERSION_GREATER_EQUAL` seems to have been added in [CMake 3.7](https://cmake.org/cmake/help/latest/release/3.7.html). Is it ok to bump the minimum version to that?

Reply:
Sure, I think that's OK (released ~4 years ago).

Reply:
Went with rearranging the if a bit because mac 10.11 only had CMake 3.6.